pkg/gadgets/trace/exec/tracer/tracer_test.go

```diff
@@ -0,0 +1,214 @@
+
+	tracer, err := tracer.NewTracer(&tracer.Config{}, nil, eventCallback)
+	if err != nil {
+		t.Fatalf("Error creating tracer: %s", err)
+	}
+	if tracer == nil {
+		t.Fatal("Returned tracer was nil")
+	}
+	t.Cleanup(func() { tracer.Stop() })
+}
+
+func TestStopTracerIdempotent(t *testing.T) {
+	utilstest.RequireRoot(t)
+
+	eventCallback := func(types.Event) {}
+
+	tracer, err := tracer.NewTracer(&tracer.Config{}, nil, eventCallback)
+	if err != nil {
+		t.Fatalf("Error creating tracer: %s", err)
+	}
+	if tracer == nil {
+		t.Fatal("Returned tracer was nil")
+	}
+	t.Cleanup(func() { tracer.Stop() })
+
+	// Check that a double stop doesn't cause issues
+	tracer.Stop()
+	tracer.Stop()
+}
+
+func TestTracer(t *testing.T) {
+	utilstest.RequireRoot(t)
+
+	// Variable to save the value of the created process PID
+	var catPid int
+
+	manyArgs := []string{}
+	// 19 is DEFAULT_MAXARGS - 1 (-1 because args[0] is on the first position).
+	for i := 0; i < 19; i++ {
+		manyArgs = append(manyArgs, "/dev/null")
+	}
+
+	// Function to generate an event used most of the times
+	generateEvent := func() error {
+		cmd := exec.Command("/bin/cat", "/dev/null")
+		if err := cmd.Run(); err != nil {
+			return err
+		}
+
+		catPid = cmd.Process.Pid
+
+		return nil
+	}
+
+	testRunner := utilstest.NewTestRunner[types.Event](t)
+
+	tests := map[string]utilstest.Test[types.Event]{
+		"test without filtering": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				tracer, err := tracer.NewTracer(&tracer.Config{}, nil, eventCallback)
+				if err != nil {
+					testRunner.Fatalf("Error creating tracer: %s", err)
+				}
+				return tracer
+			},
+			GenerateEvent: generateEvent,
+			GetAnyExpectedEvent: func(info *utilstest.Info) *types.Event {
+				return &types.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					Pid:       uint32(catPid),
+					Ppid:      uint32(info.Pid),
+					UID:       uint32(info.UID),
+					MountNsID: info.MountNsID,
+					Retval:    0,
+					Comm:      "cat",
+					Args:      []string{"/bin/cat", "/dev/null"},
+				}
+			},
+		},
+		"test filtering (no event should be captured)": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				m := utilstest.CreateMntNsFilterMap(t, 0)
+				tracer, err := tracer.NewTracer(&tracer.Config{MountnsMap: m}, nil, eventCallback)
+				if err != nil {
+					testRunner.Fatalf("Error creating tracer: %s", err)
+				}
+				return tracer
+			},
+			GenerateEvent: generateEvent,
+		},
+		"test filtering": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				m := utilstest.CreateMntNsFilterMap(t, info.MountNsID)
+				tracer, err := tracer.NewTracer(&tracer.Config{MountnsMap: m}, nil, eventCallback)
+				if err != nil {
+					testRunner.Fatalf("Error creating tracer: %s", err)
+				}
+				return tracer
+			},
```

Comment:
Since only the configuration for tracer changes, perhaps creating a tracer could be de-duplicated into a separate function or into a loop somehow?